lib/GetStream/StreamChat/StreamResponse.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace GetStream\StreamChat;
+
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A class that extends ArrayObject and additionally contains response metadata
+ * such as rate limits.
+ */
+class StreamResponse extends \ArrayObject
+{
+    /**
+     * @var StreamRateLimit|null
+     */
+    private $rateLimits;
+
+    /**
+     * @var int
+     */
+    private $statusCode;
+
+    /**
+      * @param array $array
+      * @param ResponseInterface $response
+      */
+    public function __construct($array, $response)
+    {
+        parent::__construct($array);
+
+        if ($response->hasHeader("x-ratelimit-limit")) {
+            $this->rateLimits = new StreamRateLimit(
+                $response->getHeader("x-ratelimit-limit")[0],
+                $response->getHeader("x-ratelimit-remaining")[0],
+                $response->getHeader("x-ratelimit-reset")[0]
+            );
+        }
```

Comment:
There's a small chance that one of these headers will be missing. We should ideally check presence of all of them

Reply:
@AnatolyRugalev ✅